### ISSUE TITLE
img_load requesting size (prescaling) problems

### ISSUE THREAD

**Issue by tomaszg7** [BestImageViewer/geeqie#545]
Let me start another issue for potential problem with `img_load.c`. It was written in https://github.com/BestImageViewer/geeqie/issues/541 but should probably be moved.

I've been checking some resizing things and I start to believe something is wrong in the code. I couldn't get Geeqie to request a scaled down thumbnail. There are functions which allow to set a requested size, but they don't seem to work. From my checks Geeqie always requests size 0x0. Seems that `load` part and `set_size` are called in the wrong order. Even in principle something is fishy. In order to compute requested size, Geeqie has to already know original image size (to take care of aspect ratio). And it knows that only after it gets loaded. Computed requested sizes never get used. Moreover JPG loader code later redefines requested_[width,height] values to match size of the actual picture (one of which is always wrong `w=1` - looks like a bug in libjpeg) and ignores it afterwards. Either I don't understand how it works, or something is wrong. The first choice is more probable ;) but the comparison of behaviour of GDK loader and JPEG/TIFF suggests otherwise :/ Also there is a comment in `thumb.c`:
```
/* this will speed up jpegs by up to 3x in some cases */
```
I'll try to do some speed measurements and if it turns out that the code is really dead, I'll look into fixing it or scrapping it.
